lib/rust/ensogl/component/scroll-area/src/lib.rs

```diff
@@ -127,13 +154,18 @@ impl ScrollArea {
             h_scrollbar.set_length     <+ frp.resize.map(|size| size.x);
             v_scrollbar.set_length     <+ frp.resize.map(|size| size.y);
 
-            eval frp.resize([h_scrollbar,v_scrollbar](size) {
+            eval frp.resize([h_scrollbar,v_scrollbar,mask](size) {
                 h_scrollbar.set_position_y(-size.y+scrollbar::WIDTH/2.0);
                 v_scrollbar.set_position_x(size.x-scrollbar::WIDTH/2.0);
                 h_scrollbar.set_position_x(size.x/2.0);
                 v_scrollbar.set_position_y(-size.y/2.0);
+                mask.size.set(*size);
+                mask.set_position_x(size.x / 2.0);
+                mask.set_position_y(-size.y / 2.0);
             });
```

Comment:
move `h_scrollbar`, `v_scrollbar`, and `mask` to an internal CloneRefable structure, that will have a method `update_placement` or something like that, and use it here instead. We should try keeping `frp::extend! { network ...}` small (I know its not the case in some files – they should be fixed one day).